File: 프로그래머스/0/181861. 배열의 원소만큼 추가하기/배열의 원소만큼 추가하기.js
function solution(arr) {
  let X = [];
  
  for (let i = 0; i < arr.length; i++) {
    let num = arr[i];
    X.length += num;
    
    for (let j = 0; j < num; j++) {
      X[X.length - num + j] = num;
    }
  }

  return X;
}

File: 프로그래머스/0/181861. 배열의 원소만큼 추가하기/배열의 원소만큼 추가하기.java
import java.util.ArrayList;

public class Solution {
    public static ArrayList<Integer> solution(int[] arr) {
        ArrayList<Integer> newArr = new ArrayList<>();
        
        for (int a : arr) {
            for (int i = 0; i < a; i++) {
                newArr.add(a);
            }
        }
        
        return newArr;
    }
}